docs/examples.md

```diff
@@ -5,87 +5,88 @@ of pull streams are used in simple real-world examples.
+When the sink gets a chunk, it iterates over the characters in it
+counting the `\n` characters. When the source returns `end` to the
 sink, the sink calls a user provided callback.
 
-# source that may fail. (read, err, end)
+# A source that may fail. (read, err, end)
 
-download a file over http and write it to fail.
+A file is downloaded over http and written to a file.
 The network should always be considered to be unreliable,
-and you must design your system to recover from failures.
-So there for the download may fail (wifi cuts out or something)
+and you must design your system to recover if the download
+fails. (For example if the wifi were to cut out).
 
 The read stream is just the http download, and the sink
-writes it to a tempfile. If the source ends normally,
-the tempfile is moved to the correct location.
-If the source errors, the tempfile is deleted.
-
-(you could also write the file to the correct location,
-and delete it if it errors, but the tempfile method has the advantage
-that if the computer or process crashes it leaves only a tempfile
-and not a file that appears valid. stray tempfiles can be cleaned up
-or resumed when the process restarts)
-
-# sink that may fail
-
-If we read a file from disk, and upload it,
-then it is the sink that may error.
-The file system is probably faster than the upload,
-so it will mostly be waiting for the sink to ask for more.
-usually, the sink calls read, and the source gets more from the file
-until the file ends. If the sink errors, it calls `read(true, cb)`
+writes it to a tempory file. If the source ends normally,
+the tempory file is moved to the correct location.
+If the source errors, the tempory file is deleted.
+
+(You could also write the file to the correct location,
+and delete it if it errors, but the tempory file method has the advantage
+that if the computer or process crashes it leaves only a tempory file
+and not a file that appears valid. Stray tempory files can be cleaned up
+or resumed when the process restarts.)
+
+# A sink that may fail
+
+If we read a file from disk, and upload it, then the upload is the sink that may error.
+The file system is probably faster than the upload and
+so it will mostly be waiting for the sink to ask for more data.
+Usually the sink calls `read(null, cb)` and the source retrives chunks of the file
+until the file ends. If the sink errors, it then calls `read(true, cb)`
 and the source closes the file descriptor and stops reading.
 In this case the whole file is never loaded into memory.
 
-# sink that may fail out of turn.
+# A sink that may fail out of turn.
 
 A http client connects to a log server and tails a log in realtime.
-(another process writes to the log file,
-but we don't need to think about that)
+(Another process will write to the log file,
+but we don't need to worry about that.)
 
-The source is the server log stream, and the sink is the client.
+The source is the server's log stream, and the sink is the client.
```

Comment:
Don't require the apostrophe.

Reply:
hmm, its a"log stream" that belongs to a server. Rather than there being multiple servers.

Reply:
The gritty details of grammar in technical writing :) For what it's worth, I think if you add the 's' you would want the apostrophe because it is possessive.

Reply:
Yep, sounds good to me. 👍

Reply:
👍